### PR TITLE
Add Azure jobs to CSI driver and CSO

### PR DESCRIPTION
**Pull request by jsafrane**
cluster-storage-operator will start installing Azure Disk CSI driver soon. The CSI driver is opt-in, still, let's make sure it will not break installation of a generic Azure cluster.

### CODE REVIEW

**Comment by jsafrane** [comment]
cc @openshift/storage

File: ci-operator/jobs/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -199,6 +199,73 @@ presubmits:
...
+    cluster: build01
+    context: ci/prow/e2e-azure
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+    labels:
+      ci-operator.openshift.io/prowgen-controlled: "true"
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-cluster-storage-operator-master-e2e-azure
+    optional: true
```
Comment (bertinatto) [inline ci-operator/jobs/openshift/cluster-storage-operator/openshift-cluster-storage-operator-master-presubmits.yaml:215]: :+1:

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Comment by bertinatto** [comment]
/hold
for rebase

**Review by bertinatto — COMMENTED** [review]
/hold cancel
/lgtm

**Pull request merged**